### DISCUSSION TITLE
TextArea rows attribute hardcoded as 3

### DISCUSSION THREAD

**Discussion by mannehedlund** [twilio-labs/paste#2099] [Ideas] [answered]
Hello, I'm using a TextArea but need it to be 1 row by default and auto-expandable (which it is inherently). However, the "rows" attribute is hardcoded as 3 in the TextArea component, which means I have to do some hacky things to override it. I'm wondering if making the "rows" attribute a prop is something you have considered. I believe it would make things much more flexible.

**Comment by shleewhite** [comment]
Hi @mannehedlund! TextArea having 3 rows is the intentional behavior ([see this discussion for more history](https://github.com/twilio-labs/paste/discussions/700)).

For your use case, it may actually be better to use [a content editable](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Editable_content) `<Box>`. I think this would help with the focus styles you mentioned [in the other discussion](https://github.com/twilio-labs/paste/discussions/2080).

**Comment by shleewhite** [comment, marked as answer]
@mannehedlund Came to design system office hours and we decided he should continue using his workaround until Paste has a component that is more appropriate for a chat input.
